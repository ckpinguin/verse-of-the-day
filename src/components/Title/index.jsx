import React from 'react';
import PropTypes from 'prop-types';

import './index.css';

import { isDebug, debug } from '../../debug';

Title.propTypes = {
    value: PropTypes.string.isRequired
};
function Title({ value }) {
    return (
        <div className="Title" style={ isDebug ? debug.borderStyle : {} }>
            {isDebug && <em>Title</em>}
            <h2>{value}</h2>
        </div>
    );
}
export default Title;