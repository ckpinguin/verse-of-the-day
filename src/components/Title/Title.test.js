import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme'; 
import renderer from 'react-test-renderer';

import Title from '.';

describe('<Title />', () => {
    let props;
    let mountedTitle;
    let shallowTitle;
    const title = () => {
        if (!mountedTitle) {
            mountedTitle = mount(
                <Title {...props} />
            );
        }
        return mountedTitle;
    };
    const titleShallow = () => {
        if (!shallowTitle) {
            shallowTitle = shallow(
                <Title {...props} />
            );
        }
        return shallowTitle;
    };
        
    beforeEach(() => {
        // App = require('./App').default;        
        props = {
            value: 'Our daily bread'
        };
        mountedTitle = undefined;
        shallowTitle = undefined;
    });
    
    it('renders without crashing', () => {
        ReactTestUtils.renderIntoDocument(<Title {...props} />);
    });

    it('renders shallow', () => {
        titleShallow();
    });

    it('mounts successfully (full DOM rendering)', () => {
        title();
    });

    it('allows us to set props', () => {
        const wrapper = title();
        expect(wrapper.props().value).toEqual('Our daily bread');
        wrapper.setProps({ value: 'bonka' });
        expect(wrapper.props().value).toEqual('bonka');
    });

    it('always renders exactly one div', () => {
        const divs = titleShallow().find('div'); // shallow goes only 1 level deep
        expect(divs.length).toBe(1);
    });

    it('did not change UI compared to recorded snapshot', () => {
        const component = renderer.create(<Title {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});