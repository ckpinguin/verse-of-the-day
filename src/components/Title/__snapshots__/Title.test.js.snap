// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> did not change UI compared to recorded snapshot 1`] = `
<div
  className="Title"
  style={Object {}}
>
  <h2>
    Our daily bread
  </h2>
</div>
`;
